configs/components/puppet.rb

```diff
@@ -58,15 +58,15 @@
       [<<-HERE.undent
         mkdir -p  #{rpm_statedir} && chown root #{rpm_statedir} && chmod 0700 #{rpm_statedir} || :
         if [ -x #{puppet_bin} ] ; then
-          #{puppet_bin} resource service puppet > #{service_statefile} || :
+          #{puppet_bin} resource service puppet | grep ensure | sed 's/ensure => //' | sed s'/.$//' | sed 's/^..//' | sed "s/'//g" > #{service_statefile} || :
```

Comment:
This can be tightened up to just:

`#{puppet_bin} resource service puppet | grep -Pzo "(?<=ensure => ').*(?=')"`

Reply:
We use this same pre-inst step for AIX. Do all those grep flags work correctly there?

My inclination would be a `grep ensure | awk <some expr>`, to ensure compatibility with weirder platforms that use RPM